### PR TITLE
moved logic of remove hidden values and types to updateNaclFiles func

### PR DESCRIPTION
**Pull request by orendig**

### CODE REVIEW

File: packages/workspace/src/workspace/workspace.ts [hunk 1]
```diff
@@ -31,8 +33,12 @@ import { Errors, ServiceDuplicationError, EnvDuplicationError,
 import { EnvConfig } from './config/workspace_config_types'
 import {
   addHiddenValuesAndHiddenTypes,
+  removeHiddenValuesForInstance,
 } from './hidden_values'
 import { WorkspaceConfigSource } from './workspace_config_source'
+import {
+  createAddChange, createRemoveChange,
```
Comment (roironn) [inline packages/workspace/src/workspace/workspace.ts:40]: We should probably move these functions out of projections if they are used here

File: packages/workspace/src/workspace/workspace.ts [hunk 2]
```diff
@@ -136,6 +142,71 @@ export const loadWorkspace = async (config: WorkspaceConfigSource, credentials:
...
+      )
+    }
+    return change
+  }
+
+  const updateNaclFiles = async (
+    changes: DetailedChange[],
+    mode?: RoutingMode
+  ): Promise<void> => {
+    const changesAfterHiddenRemoved = await Promise.all(changes.filter(
```
Comment (roironn) [inline packages/workspace/src/workspace/workspace.ts:202]: I think we should probably wrap this and all of the functions called by this to hidden_types.ts DYT?

File: packages/workspace/src/workspace/workspace.ts (hunk 1 above)
Comment (orendig) [reply packages/workspace/src/workspace/workspace.ts:40]: I think we can move them to a new (util/helper) class, WDYT?

File: packages/workspace/src/workspace/workspace.ts (hunk 2 above)
Comment (orendig) [reply packages/workspace/src/workspace/workspace.ts:202]: Actually we have here 2 options: your suggestion or to move all (2) funcs of hidden_types.ts to workspace.ts. wdyt better?
@roironn

File: packages/core/src/api.ts [hunk 3]
```diff
@@ -145,9 +140,10 @@ export const deploy = async (
   }
   const errors = await deployActions(actionPlan, adapters, reportProgress, postDeployAction)
 
-  // Remove hidden Types and hidden values inside instances
-  const elementsAfterHiddenRemoval = removeHiddenValuesAndHiddenTypes(changedElements.values())
-    .map(e => e.clone())
+  // Clone the elements because getDetailedChanges can change its input
```
Comment (ori-moisis) [inline packages/core/src/api.ts:143]: is this a new thing?
we use `getDetailedChanges` in a bunch of places, if it started changing its input we should fix it so it doesn't (preferably) or make sure it is ok in all the other places that use it

Comment (orendig) [reply packages/core/src/api.ts:143]: @ori-moisis are you asking about the cloning? The usage of clone isn't new, and originally you wrote this comment :)
I think (after another view over the code) that the clone is redundant and getDetailedChanges logic out of place.

**Pull request merged**